ouroboros-network/src/Ouroboros/Network/PeerSelection/LedgerPeers.hs

```diff
@@ -255,20 +259,33 @@ ledgerPeersThread PeerActionsDNS {
                  traceWith wlpTracer DisabledLedgerPeers
                  return (Map.empty, Map.empty, now)
                UseLedgerPeers ula -> do
-                 peers <- (\case
-                            BeforeSlot          -> []
-                            LedgerPeers _ peers -> peers
-                          )
-                      <$> atomically (getLedgerPeers wlpConsensusInterface ula)
-                 let peersStake   = accPoolStake peers
-                     bigPeersStakeMap = accBigPoolStakeMap peers
+                 (ledgerStateJudgement, consensusSlotNo, consensusPeers, peerSnapshot) <-
+                   atomically ((,,,) <$> lpGetLedgerStateJudgement
+                                     <*> lpGetLatestSlot
+                                     <*> getLedgerPeers wlpConsensusInterface ula
+                                     <*> wlpGetLedgerPeerSnapshot)
+
+                 -- we have to assess which of, if any, peers we get from consensus vs.
+                 -- peers we may have from the snapshot file is more recent, and use that
+                 let (accPoolStake -> peersStake, bigPeersStakeMap) =
+                       case (consensusSlotNo, consensusPeers, peerSnapshot) of
+                         (At t, LedgerPeers _ lp, Just (LedgerPeerSnapshot (At t', sp {- snapshot peer-})))
+                           | t' > t -> (lp, Map.fromAscList sp)
+                           | otherwise -> (lp, accBigPoolStakeMap lp)
```

Comment:
I wish there would be some symmetry between this type and `LedgerPeers` that encapsulates the Before/After slot semantics. You could perhaps add the slot info and have a semigroup operation that would just keep the most recent :)

Reply:
That's a neat idea, but will it work? I have to treat the snapshot peers a little differently from than the peers I get from consensus, because the snapshot peers were already computed to contain just the big ledger peers, and I don't want to compute big ledger peers of the big ledger peers - I use the function Map.fromAscList to retrieve the snapshot peers vs. accPoolStake/accBigPoolStake for the ones I get from the ledger. Maybe if I throw some tag in there as well to distinguish... On the other hand, do you have an idea whether we could use this Semigroup instance or if we could generalize over this pattern somewhere else in the code? It seems like it is just one-off and so having it 'inlined' here like I did may be actually clearest, but maybe I'm not seeing something.

Reply:
Ahh that's right I forgot that small detail, given that I don't think it is worth then